### ISSUE TITLE
Incorrect scope checks of class properties (Regression)

### ISSUE THREAD

**Issue by falsandtru** [Microsoft/TypeScript#14208]
**TypeScript Version:**  master

**Code**

```ts
class C {
  constructor() {
    new D().p; // ok
  }
  public p = new D().p; // should also be ok
}
class D {
  public p = '';
}
```

**Expected behavior:**

ok

**Actual behavior:**

```
$ node built/local/tsc.js index.ts -t es6
index.ts(5,22): error TS2448: Block-scoped variable 'p' used before its declaration.
```

**Comment by 82antu** [comment]
Typescript 2.2.1, same Issue

**Comment by falsandtru** [comment]
@sandersn Could you fix this?

**Comment by sandersn** [comment]
Fix is up at #14952

**Comment by falsandtru** [comment]
Thanks!
